scripts/dependency-diagram.py

```
@@ -0,0 +1,200 @@
+    for src, dst in edges:
+        nodes.add(src)
+        nodes.add(dst)
+
+    order = topo_order(edges)
+
+    # Group utils crates so Mermaid places them at the top.
+    utils = sorted(n for n in nodes if n.startswith("utils-"))
+
+    lines = ["```mermaid"]
+    if title:
+        lines.append(f"---")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)